proto/v1/compliance.proto

```diff
@@ -0,0 +1,71 @@
+// Copyright 2019 The Grafeas Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package grafeas.v1;
+
+import "google/protobuf/timestamp.proto";
+import "proto/v1/cvss.proto";
+import "proto/v1/package.proto";
+import "proto/v1/vulnerability.proto";
+
+option go_package = "github.com/grafeas/grafeas/proto/v1/grafeas_go_proto";
+option java_multiple_files = true;
+option java_package = "io.grafeas.v1";
+option objc_class_prefix = "GRA";
+
+message ComplianceNote {
+  // The OS and config versions the benchmark applies to.
+  repeated grafeas.v1.ComplianceVersion version = 2;
+  // The title that identifies this compliance check.
+  string title = 3;
+  // A description about this compliance check. 
+  string description = 4;
+  // A rationale for the existence of this compliance check.
+  string rationale = 5;
+  // A description of remediation steps if the compliance check fails.
+  string remediation = 6;
+  // A compliance check that is a CIS benchmark.
+  message CisBenchmark {
+    int32 profile_level = 1;
+    grafeas.v1.Severity severity = 2;
+  }
+  oneof compliance_type {
+    CisBenchmark cis_benchmark = 7;
+  }
+  // Serialized scan instructions with a predefined format.
+  bytes scan_instructions = 8;
+}
+
+// Describes the CIS benchmark version that is applicable to a given OS and 
+// os version. 
+message ComplianceVersion {
+  // The CPE URI (https://cpe.mitre.org/specification/) this benchmark is
+  // applicable to.
+  string cpe_uri = 1;
+  // The version of the benchmark. This is set to the version of the OS-specific
+  // CIS document the benchmark is defined in.
+  string version = 2;
+}
+
+message ComplianceOccurrence {
+  repeated NonCompliantFile non_compliant_files = 2;
+}
+message NonCompliantFile {
+  string path = 1; // Empty if `display_command` is set.
```

Comment:
what does `display_command` mean in the context of files? i.e. why won't the path be set if the `display_command` is set? this seems like something an implementation could decide to enforce or not

Reply:
display_command is the command that can be used to display the non-compliant files. For example, for the benchmark "6.1.10 Ensure no world writable files exist", we could list all files on the filesystem that were found to be world-writable:
```
[{path: "/home/bob/file1.txt", reason: "File's +w bit set for all users."},
 {path: "/home/bob/file2.txt", reason: "File's +w bit set for all users."},
 {path: "/home/bob/file3.txt", reason: "File's +w bit set for all users."},
 ...]

```
But we could also just show this one command:
```
{display_command: "df --local -P | awk '{if (NR!=1) print $6}'
        | xargs -I '{}' find '{}' -xdev -type f -perm -0002",
 reason: "One or more files have the +w bit set for all users."}
```

When this command is set, it doesn't make much sense to list the paths individually since the command already defines the files.

Reply:
Got it, thank you! Would it be possible to capture some of this explanation in a brief comment, for clarity?

Reply:
Done.